### PR TITLE
Arrowhead exercise v2

### PR DESCRIPTION
**Pull request by mostlybadfly**
Hello! I wanted to submit this to get some feedback after the initial suggestions.  I can see how much cleaner the code looks now, any other suggestions would be appreciated :smile:

### CODE REVIEW

File: arrowhead/arrowhead.rb [hunk 1]
```diff
@@ -15,19 +15,20 @@ class Arrowhead
...
-    else
-      raise "Unknown region, please provide a valid region."
-    end
+    shapes = check_region(region)
+    arrowhead = check_shape(shapes, shape)
+    "You have a(n) '#{arrowhead}' arrowhead. Probably priceless."
+  end
+
+  def self.check_region(region)
+    raise "Uknown region, please provide a valid region." unless CLASSIFICATIONS.include? region
```
Comment (ochagata) [inline arrowhead/arrowhead.rb]: You can create custom errors that have the messages in them, that way you don't have to write the same string over and over again.

File: arrowhead/arrowhead.rb [hunk 2]
```diff
@@ -15,19 +15,20 @@ class Arrowhead
...
+    CLASSIFICATIONS[region]
+  end
+
+  def self.check_shape(shapes, shape)
```
Comment (ochagata) [inline arrowhead/arrowhead.rb]: This is a method that returns a boolean value, thus it should have a question mark at the end.

Comment (ochagata) [inline arrowhead/arrowhead.rb]: Also, consider renaming to something more descriptive: `valid_shape?`

File: arrowhead/arrowhead.rb [hunk 3]
```diff
@@ -15,19 +15,20 @@ class Arrowhead
...
-      end
-    else
-      raise "Unknown region, please provide a valid region."
-    end
+    shapes = check_region(region)
+    arrowhead = check_shape(shapes, shape)
+    "You have a(n) '#{arrowhead}' arrowhead. Probably priceless."
+  end
+
+  def self.check_region(region)
```
Comment (ochagata) [inline arrowhead/arrowhead.rb]: A bit confusing with the name of this method, it sounds like it should be checking, but returns a hash as an output. Consider renaming

File: arrowhead/arrowhead.rb [hunk 4]
```diff
@@ -15,19 +15,20 @@ class Arrowhead
...
+    raise "Uknown region, please provide a valid region." unless CLASSIFICATIONS.include? region
+    CLASSIFICATIONS[region]
+  end
+
+  def self.check_shape(shapes, shape)
+    raise "Uknown shape value. Are you sure you know what you're talking about?" unless shapes.include? shape
```
Comment (ochagata) [inline arrowhead/arrowhead.rb]: Same here for the error message

File: arrowhead/arrowhead.rb (hunk 2 above)
Comment (ochagata) [inline arrowhead/arrowhead.rb]: Consider refactoring this method so it returns a bool value

**Comment by ochagata** [comment]
I left a few comments, otherwise it looks fine

**Comment by mostlybadfly** [comment]
Thanks for the suggestions @ochagata :) . I moved the error messages to their own methods and realized I didn't need to return from them so the names changed.  Going to move on to the next exercise.

**Comment by paulghaddad** [comment]
@mostlybadfly Nice work! As Ozzy said, you are free to move on.

**Pull request closed without merging**